src/middleware/redirect/mod.rs

```diff
@@ -91,14 +91,24 @@ impl Middleware for Redirect {
         // and try sending it until we get some status back that is not a
         // redirect.
 
+        let base_url = req.url().clone();
+
         while redirect_count < self.attempts {
             redirect_count += 1;
             let r: Request = req.clone();
             let res: Response = client.send(r).await?;
             if REDIRECT_CODES.contains(&res.status()) {
                 if let Some(location) = res.header(headers::LOCATION) {
                     let http_req: &mut http::Request = req.as_mut();
-                    *http_req.url_mut() = Url::parse(location.last().as_str())?;
+                    *http_req.url_mut() = match Url::parse(location.last().as_str()) {
+                        Ok(valid_url) => valid_url,
```

Comment:
This is not quite complete, a redirect could give us a full url and then a future one could give us a partial one.

I think this branch should also update the`base_url`.